### ISSUE TITLE
Azuerm_eventhub: Namespace Premium capacity does not scale PU

### ISSUE THREAD

**Issue by martin-4screen** [hashicorp/terraform-provider-azurerm#17897]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

latest (1.2.6)

### AzureRM Provider Version

3.16.0

### Affected Resource(s)/Data Source(s)

azuerm_eventhub

### Terraform Configuration Files

```hcl
resource "azuerm_eventhub_namespace" "someEventNamespace" {
  name                = join("", [var.azure_event_hub_name, replace(var.environment, "-", "")])
  location            = var.location
  resource_group_name = var.resource_group_name
  sku                 = "Premium"
  zone_redundant      = true
  capacity            = 2

  network_rulesets {
    default_action = "Deny"
    virtual_network_rule = [
      {
        subnet_id                                       = var.subnet_id
        ignore_missing_virtual_network_service_endpoint = false
      }
    ]
  }
}
```

### Debug Output/Panic Output

```shell
# module.messaging.azurerm_eventhub_namespace.someEventNamespace will be updated in-place
  ~ resource "azurerm_eventhub_namespace" "someEventNamespace" {
      ~ capacity                            = 1 -> 2
        id                                  = "/subscriptions/***/resourceGroups/SomeRGDev/providers/Microsoft.EventHub/namespaces/someEventNamespaceDev"
        name                                = "someEventNamespaceDev"
        tags                                = {}
        # (11 unchanged attributes hidden)
    }

...


Namespace Name: "someEventNamespaceDev"): namespaces.NamespacesClient#Update: Failure sending request: StatusCode=409 -- Original Error: autorest/azure: Service returned an error. Status=<nil> Code="Conflict" Message="<Error><Code>409</Code><Detail>Throughput units on the Namespace: 'someEventNamespaceDev' cannot be decreased, as it results into creating more than allowed number of cumulative partitions. This namespace has: '97' partitions and maximum number of partitions allowed per throughput unit is: '40', which implies minimum number of throughput units should be: '3'. Deleting Event Hubs can bring down the cumulative partition count.</Detail></Error>
```

### Expected Behaviour

The amount of processing units on the azure event hub namespace should scale from 1 -> 2. 

### Actual Behaviour

The amount of the processing units stays the same. Furthermore, the premium namespace should come along with over [200 partitions ](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-quotas) , but i am getting a failure message regarding the `maximum number of partitions allowed per throughput unit` indicating, that there seems to be some overlapping with the other subscription models ( like standard). 

I also tried to set `auto_inflate_enabled = true` and set a maximum for the throughput units, but I encountered the same results as can be seen in the panic output. 

### Important Factoids

Azure Germany

**Comment by lhaussknecht** [comment]
I just tried the same but my error message looks a bit different:

```
  # module.eventhub["xxx"].azurerm_eventhub_namespace.eventhub will be updated in-place
  ~ resource "azurerm_eventhub_namespace" "eventhub" {
      ~ capacity                            = 1 -> 2
        id                                  = "/subscriptions/***/resourceGroups/xxx/providers/Microsoft.EventHub/namespaces/xxx"
        name                                = "xxx"
        tags                                = {}
        # (11 unchanged attributes hidden)

        # (2 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
╷
│ Error: creating Namespace (Subscription: "***"
│ Resource Group Name: "xxxx"
│ Namespace Name: "xxxx"): performing CreateOrUpdate: namespaces.NamespacesClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code="BadRequest" Message="Encryption properties are missing. Note: Once encryption has been enabled on a namespace, it cannot be disabled. CorrelationId: xxx
```

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of #18542. As such, I am going to mark this issue as closed. If that is not the case, please provide additional information including the version in which you are still experiencing this issue, thanks!
